jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/ServletChannelState.java

```diff
@@ -102,12 +103,45 @@ private enum RequestState
 
     /*
      * The input readiness state.
+     * <pre>
+     *              read() without preceding
+     *              isReady()  ------
+     *                         \     \   unhandle() returns Action.READ_CALLBACK to call the ReadListener,
+     *                          \     \  or read() stole available content after setReadListener()
+     *                           --> IDLE <--------------
+     *     blocking read() unblocked  ^                  \
+     *                                |                   \
+     *                                |                    \  setReadListener() called while
+     *            registering demand  v                     v content is available
+     *                               UNREADY ------------> READY
+     *                                         demand
+     *                                         serviced
+     * </pre>
      */
     private enum InputState
     {
-        IDLE,       // No isReady; No data
-        UNREADY,    // isReady()==false; No data
-        READY       // isReady() was false; data is available
+        /**
+         * The 'default' state, when there is no pending demand nor a pending notification to the ReadListener.
+         * There are 3 ways to transition to this state:
+         * <li>from IDLE: when an async read() is called without a preceding call to isReady()</li>
```

Comment:
`<li>` without surrounding `<ul>` also in the other javadocs.

Reply:
Fixed.